### PR TITLE
kubevela 1.1.13

### PR DESCRIPTION
**Pull request by sunny0826**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by sunny0826** [comment]
/cc @chenrui333 @carlocab   
Can you help merge this PR?

File: Formula/kubevela.rb [hunk 1]
```diff
@@ -18,8 +18,13 @@ class Kubevela < Formula
   depends_on "go" => :build
 
   def install
```
Comment (carlocab) [inline Formula/kubevela.rb:20]: Any reason we're not setting `CGO_ENABLED=0` here, as in the [`Makefile`](https://github.com/oam-dev/kubevela/blob/f7ebbf4fa0f036a2d3a9ad90f8adba78f0a804bd/Makefile#L12)?

File: Formula/kubevela.rb [hunk 2]
```diff
@@ -18,8 +18,13 @@ class Kubevela < Formula
...
-    system "make", "vela-cli", "VELA_VERSION=#{version}"
-    bin.install "bin/vela"
+    ldflags = %W[
+      -s -w
+      -X github.com/oam-dev/kubevela/version.VelaVersion=#{version}
+      -X github.com/oam-dev/kubevela/version.GitRevision=#{Utils.git_head}
+    ].join(" ")
```
Comment (carlocab) [inline Formula/kubevela.rb:25]:
```suggestion
    ]
```

File: Formula/kubevela.rb [hunk 3]
```diff
@@ -18,8 +18,13 @@ class Kubevela < Formula
...
+
+    system "go", "build", *std_go_args(ldflags: ldflags), "-o", bin/"vela", "./references/cmd/cli"
```
Comment (carlocab) [inline Formula/kubevela.rb:27]:
```suggestion
    system "go", "build", *std_go_args(output: bin/"vela", ldflags: ldflags), "./references/cmd/cli"
```

File: Formula/kubevela.rb [hunk 4]
```diff
@@ -18,8 +18,13 @@ class Kubevela < Formula
...
   end
 
   test do
```
Comment (carlocab) [inline Formula/kubevela.rb:30]: Can we add a test to check the version given that we're passing it via `ldflags` now?

Comment (sunny0826) [reply Formula/kubevela.rb:32]: done

File: Formula/kubevela.rb (hunk 2 above)
Comment (sunny0826) [reply Formula/kubevela.rb]: fix

File: Formula/kubevela.rb (hunk 1 above)
Comment (sunny0826) [reply Formula/kubevela.rb:20]: added 
```ruby
ENV["CGO_ENABLED"] = "0"
ENV["GO111MODULE"] = "on"
```

File: Formula/kubevela.rb [hunk 5]
```diff
@@ -18,18 +18,20 @@ class Kubevela < Formula
   depends_on "go" => :build
 
   def install
+    ENV["CGO_ENABLED"] = "0"
+    ENV["GO111MODULE"] = "on"
```
Comment (carlocab) [inline Formula/kubevela.rb:22]:
```suggestion
```
This is the default.

File: Formula/kubevela.rb [hunk 6]
```diff
@@ -18,18 +18,20 @@ class Kubevela < Formula
...
-    ].join(" ")
+    ]
 
-    system "go", "build", *std_go_args(ldflags: ldflags), "-o", bin/"vela", "./references/cmd/cli"
+    system "go", "build", *std_go_args(output: bin/"vela", ldflags: ldflags), "./references/cmd/cli"
   end
 
   test do
     # Should error out as vela up need kubeconfig
-    status_output = shell_output("#{bin}/vela up 2>&1", 1)
```
Comment (carlocab) [inline Formula/kubevela.rb:27]: We should keep this check.

File: Formula/kubevela.rb [hunk 7]
```diff
@@ -18,18 +18,20 @@ class Kubevela < Formula
...
+    status_output = shell_output("#{bin}/vela version 2>&1", 1)
```
Comment (carlocab) [inline Formula/kubevela.rb:34]: We should check that `vela version` produces the correct version number too.

Comment (sunny0826) [reply Formula/kubevela.rb:34]: `get kubeConfig err invalid configuration: no configuration has been provided`  This error message is returned every time due to a built-in check.

Comment (carlocab) [reply Formula/kubevela.rb:34]: Is there no way to get `vela` to show the version number?

Comment (sunny0826) [reply Formula/kubevela.rb:34]: Maybe I can try.

**Comment by sunny0826** [comment]
@carlocab Version check test has been added, please review.

**Comment by sunny0826** [comment]
@carlocab @chenrui333  The KubeVela 1.2.x release is coming soon, can you help merge this PR before then?

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
@sunny0826 sorry about the delay, lgtm

**Comment by sunny0826** [comment]
> @sunny0826 sorry about the delay, lgtm

Thank you so much

**Pull request closed without merging**